docs/fsharp/tutorials/getting-started/getting-started-vscode.md

```diff
@@ -1,4 +1,278 @@
+
+## Making your script code part of your source code
+
+The previous sections in this article demonstrated a common first step in writing F# code: writing an initial function and executing it interactively with FSI.  This is known as REPL-driven development, where REPL stands for "Read-Evaluate-Print Loop".  It's a great way to tweak with functionality until you have something working.
+
+The next step is to incorporate code in your F# script file into F# source code.  To begin, open `ClassLibraryDemo.fs`.  You'll notice that some code is already in there.  Go ahead and delete the class definition, but make sure to leave the [`namespace`](../../language-reference/namespaces.md) declaration at the top.
+
+Next, create a new [`module`](../../language-reference/modules.md) called `PigLatin` and copy the `toPigLatin` function into it as such:
+
+[!code-fsharp[ToPigLatin](../../../../samples/snippets/fsharp/getting-started/pig-latin.fs#L1-L14)]
+
+This is one of the many ways you can organize functions in F# code, and a common approach if you also want to call your code from C# or Visual Basic projects.
+
+Next, open the `Script.fsx` file again, and delete the entire `toPigLatin` function, but make sure to keep the following two lines in the file:
+
+```
+#load "ClassLibraryDemo.fs"
+open ClassLibraryDemo
+```
+
+Those two lines will load the `ClassLibraryDemo.fs` file and `open` the `ClassLibraryDemo` namespace, just like if you were to reference this code in another project.  Note that you can also open FSI with `>FSI: Start`, then type those two lines with `;;` as the delimiter to accomplish the same thing.
+
+Next, call the function with the `PigLatin` module that you defined earlier:
+
+```
+> PigLatin.toPigLatin "banana";;
+val it : string = "ananabay"
+> PigLatin.toPigLatin "apple";;
+val it : string = "appleyay"
+```
+
+Success!  You get the same results as before, but now loaded from an F# source file.  The major difference here is that F# source files are compiled into assemblies which can be executed anywhere, not just in FSI.
+
+## Summary
+
+In this article, you've learned:
+
+1. How to set up Visual Studio Code with Ionide.
+2. How to create your first F# project with Ionide.
+3. How to use F# Scripting to write your first F# function in Ionide and then execute it in FSI.
+4. How to migrate your script code to F# source and then call that code from FSI.
+
+You're now equipped to write much more F# code using Visual Studio Code and Ionide.
+
+## Next steps
+
+To learn more about F# and the features of the language, check out [Tour of F#](../../tour.md).
+
```

Comment:
It might be useful to have a brief troubleshooting section that says how to check the fsharp language service output channel or the developer console tools in vscode.  Make note of some common early pitfalls like - 
* a workspace folder needs to be open and the script or module file saved before the editor features will kick in.  
* Remember to restart vscode if you've just made changes to your system/installed ionide prereqs. 
* check that you can use fsc and fsi without a qualified path from the command line
* your projects or directories may be spazzing out if you try to use an invalid chars/name for the project or directory, I forget what it does when this goes wrong , @Krzysztof-Cieslak should know 
* if the commands aren't working be sure to check your keybindings in the user settings
* if all else fails nuke the ionide-fsharp dir in your vscode plugin directory and reinstall


Or just link to https://github.com/ionide/ionide-vscode-fsharp#how-to-get-logs-for-debugging--issue-reporting and the gitter channel https://gitter.im/ionide/ionide-project so people know where to get help